gateway/cert_test.go

```diff
@@ -303,6 +357,7 @@ func testAPIMutualTLSHelper(t *testing.T, skipCAAnnounce bool) {
 		globalConf.HttpServerOptions.UseSSL = true
 		globalConf.HttpServerOptions.SSLCertificates = []string{certID}
 		globalConf.HttpServerOptions.SkipClientCAAnnouncement = skipCAAnnounce
+		globalConf.ControlAPIPort = 1212
```

Comment:
added this to simulate a well configured control api. A test will be added to handle situations as: all apis are mtls and control not configured, all apis are mtls and control is configured, not all apis are mtls and control not configured, etc....

this change is required as the error returned when control api is not configured is different than the one written in the tests. I was thinking in just adding 2 more values to the test matrix, but this test is currently too big and I would like to avoid adding more complexity to it.